rust/index/src/hnsw_provider.rs

```diff
@@ -304,13 +334,24 @@ impl HnswIndexProvider {
                         bytes_read,
                         key,
                     );
-                    Ok(buf)
+                    buffers.push(buf);
+                    Ok(())
                 })
                 .await?;
-            let file_path = index_storage_path.join(file);
-            self.copy_bytes_to_local_file(&file_path, buf).await?;
         }
-        Ok(())
+        match hnswlib::HnswData::new_from_buffers(
+            buffers[0].clone(),
```

Comment:
This will copy the underlying data right? That can be quite large. You can use the builder pattern above / drain() to prevent that. Would prefer we didn't clone here.